app/components/billing/address.js

```diff
@@ -1,18 +1,32 @@
 import Component from '@ember/component';
 import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
 import { countries } from 'travis/utils/countries';
 
 export default Component.extend({
   countries,
+  flashes: service(),
   classNames: ['address'],
   openEditContactForm: false,
   openEditBillingForm: false,
 
   editContact: task(function* () {
-    yield this.subscription.save();
-    this.closeEditForms();
+    try {
+      yield this.subscription.save();
+      this.closeEditForms();
+      this.flashes.clear();
+    } catch (e) {
+      this.handleErrors(e);
+    }
   }).drop(),
 
+  handleErrors(e) {
+    if (e.errors.length) {
+      const error = e.errors.firstObject;
+      this.flashes.error(error.detail);
+    }
```

Comment:
It's a good question, I can see the benefits and drawbacks of either approach. I think I would generally lean towards trusting the backend and providing more detail, but I'd be interested to hear other opinions.

Reply:
There could be other errors coming from API, e.g. `401`, `500` and they all seems to be processed by this handler. I guess, some of them might be filtered by the condition, but maybe some could pass? I'm more in favor of having a full control over what's displayed to user. Also API may change and it will pass unnoticed. 

Another useful pattern when having all the text in the app, you could easily copy a chunk of text in the app and search for it in the code, easily finding occurrences. 

By the way, I think that network error may break this error handler, because, most likely, it will have `errors` undefined. Try to switch to `offline` in Network panel of Chrome DevTools while on the page and the click to the button that runs the task. You'll see the network error in action.